addons/purchase/models/account_invoice.py

```diff
@@ -410,6 +410,10 @@ def _find_and_set_purchase_orders(self, po_references, partner_id, amount_total,
         """
         self.ensure_one()
 
+        if self.line_ids.purchase_line_id.order_id:
```

Comment:
This seems overly strong to me, as a bill could be linked to multiple purchase orders. With this change, as soon as a single line would be linked to a PO, then nothing would be matched anymore :confused: